File: testdata/fixtures/java-clean/src/main/java/com/example/domain/Order.java
// Domain layer: pure business types. Only the platform stdlib is allowed here.
package com.example.domain;

import java.util.concurrent.atomic.AtomicLong;

public final class Order {
    private static final AtomicLong NEXT_ID = new AtomicLong(1);

    public final long id;
    public final int total;

    public Order(int total) {
        this.id = NEXT_ID.getAndIncrement();
        this.total = total;
    }

    @Override
    public String toString() {
        return "order " + id + " (" + total + ")";
    }
}

File: testdata/fixtures/ruby-clean/domain/order.rb
# Domain layer: pure business types. Only the platform stdlib is allowed here.
require "securerandom"

module Domain
  class Order
    attr_reader :id, :total

    def initialize(total)
      @id = SecureRandom.uuid
      @total = total
    end
  end

  def self.new_order(total)
    Order.new(total)
  end
end
